addons/mrp_account/models/mrp_production.py

```diff
@@ -28,18 +28,27 @@ def _cal_price(self, consumed_moves):
         """
         super(MrpProduction, self)._cal_price(consumed_moves)
         work_center_cost = 0
-        finished_move = self.move_finished_ids.filtered(lambda x: x.product_id == self.product_id and x.state not in ('done', 'cancel') and x.quantity_done > 0)
+        finished_move = self.move_finished_ids.filtered(lambda m: m.product_id == self.product_id and m.state not in ('done', 'cancel') and m.quantity_done > 0)
         if finished_move:
             finished_move.ensure_one()
             for work_order in self.workorder_ids:
                 time_lines = work_order.time_ids.filtered(lambda x: x.date_end and not x.cost_already_recorded)
                 duration = sum(time_lines.mapped('duration'))
                 time_lines.write({'cost_already_recorded': True})
                 work_center_cost += (duration / 60.0) * work_order.workcenter_id.costs_hour
+            qty_done = finished_move.product_uom._compute_quantity(finished_move.quantity_done, finished_move.product_id.uom_id)
+            extra_cost = self.extra_cost * qty_done
+            total_cost = (sum([-m.stock_valuation_layer_ids.value for m in consumed_moves.sudo()]) + work_center_cost + extra_cost)
+            byproduct_moves = self.move_finished_ids.filtered(lambda m: m.product_id != self.product_id and m.state not in ('done', 'cancel') and m.quantity_done > 0)
```

Comment:
instead of using `self.move_finished_ids.filtered(lambda m: m.product_id != self.product_id)` to get byproduct line
you can use the compute field `move_byproduct_ids`, it is more readable.